plugins/foreman_salt/13.0/index.md

```diff
@@ -0,0 +1,444 @@
+
+![](/static/images/plugins/foreman_salt/states_index.png)
+
+Foreman provides an interface for adding Salt states to either a Foreman host group or a host. Salt will know to apply these states through the external node classification feature. This is done in addition to top.sls.
+
+In the example below, the inherited states are coming from the host's host group, and you may add additional states specifically to this new host:
+
+![](/static/images/plugins/foreman_salt/states.png)
+
+## 4.3 Key Management
+
+This plugin provides a UI interface to both Salt Keys and the Autosign entries.  Navigate to the Smart Proxy's page, and click on 'Salt Keys' or 'Salt Autosign' next to the Smart Proxy in the action buttons.
+
+### 4.3.1 Autosign
+
+Foreman manages the lifecycle of a salt minion's keys.  When the minion provisions, the autosign record will be added automatically, and when the build finishes it is removed.  You may also manually enter an autosign entry, for example, to support automatically signing an entire domain managed outside foreman (e.g. '\*.example.com').
+
+![](/static/images/plugins/foreman_salt/autosign.png)
+
+### 4.3.2 Keys
+
+The Salt Keys interface allows you to manually accept, reject, or delete keys.  This is useful for managing hosts outside of Foreman's orchestration.
+
+![](/static/images/plugins/foreman_salt/key_management.png)
+
+## 4.4 Highstate
+
+There are two ways to trigger `state.highstate` on Foreman:
+
+  - use the "Run Salt" button (old behavior) which will use the shell command 'salt' on the smart proxy
+  - use Schedule Remote Job / Run Salt via Foreman Remote Execution (Foreman Remote Execution Plugin needs to be installed)
+
+Results of a highstate run will show up in Monitor / Reports if using the upload-salt-reports feature of the Smart Proxy.
+
+You can also review executed state.highstate runs with the jobs runner on the Salt Master:
+
+    salt-run jobs.lookup_jid 20140921213537978546
+
+### 4.4.1 Via Schedule Remote Job / Run Salt
+
+In case the foreman_remote_execution plugin is installed, the Salt `state.highstate` can be started via Schedule Remote Job / Run Salt.
+
+![](/static/images/plugins/foreman_salt/rex.png)
+
+The "Run Salt" button wouldn't be visible in this case on a newly deployed Foreman. 
+Disabling the "Administer / Settings / Salt" setting `salt_hide_run_salt_button` would activate the "Run Salt" button again. 
+
+### 4.4.2 Via Run Salt button
+
+"Run Salt" button on the hosts page will trigger a `state.highstate` run.
+
+![](/static/images/plugins/foreman_salt/runsalt.png)
+
+## 4.5 Salt with Foreman Remote Execution Plugin
+
+Currenly, there are 2 ways to run salt via the remote execution plugin:
+
+  - Via a SSH tunnel: in this case SSH authentication needs to work. In the end, 'salt-call' on the smart proxy will be used to run the salt command.
```

Comment:
From [salt docs](https://docs.saltstack.com/en/latest/ref/cli/salt-call.html):
> The salt-call command is used to run module functions locally on a minion instead of executing them from the master.

I think proxy connects to the target host over ssh and runs `salt-call` there, no?

Reply:
right. thanks. to many options.

Reply:
fixed @adamruzicka